lib/src/widget/nav_bar/widget/nav_bar.dart

```diff
@@ -0,0 +1,133 @@
+import 'package:flutter/material.dart';
+import 'package:xayn_design/src/widget/nav_bar/widget/nav_bar_item/back_button.dart';
+import 'package:xayn_design/src/widget/nav_bar/widget/nav_bar_item/edit.dart';
+import 'package:xayn_design/src/widget/nav_bar/widget/nav_bar_item/icon_button.dart';
+import 'package:xayn_design/xayn_design.dart';
+
+class NavBar extends StatefulWidget {
+  final NavBarConfig config;
+
+  const NavBar(
+    this.config,
+  ) : super(key: null);
+
+  @override
+  _NavBarState createState() => _NavBarState();
+}
+
+class _NavBarState extends State<NavBar> {
+  Linden get linden => UnterDenLinden.getLinden(context);
+
+  bool get isBackBtn =>
+      widget.config.items.length == 1 &&
+      widget.config.items.first is NavBarItemBackButton;
+
+  @override
+  Widget build(BuildContext context) {
+    final items = widget.config.items.map(_buildItem).toList(growable: false);
+
+    if (isBackBtn) {
+      return Row(children: [items.first]);
+    }
+
+    final row = Row(
+      mainAxisSize:
+          widget.config.isWidthExpanded ? MainAxisSize.max : MainAxisSize.min,
+      mainAxisAlignment: widget.config.isWidthExpanded
+          ? MainAxisAlignment.spaceBetween
+          : MainAxisAlignment.start,
+      children: items,
+    );
+    final double internalPadding = linden.dimen.unit;
+    final double externalPadding = linden.dimen.unit2;
+
+    final sized = SizedBox(
+      height: linden.dimen.unit8,
+      child: AppCardWidget(
+        contentPadding: EdgeInsets.symmetric(horizontal: internalPadding),
+        child: row,
+        borderRadius: BorderRadius.all(Radius.circular(linden.dimen.unit1_5)),
+      ),
+    );
+    final maxWidth = linden.dimen.isMobilePortraitMode
+        ? linden.dimen.screenWidth - (internalPadding) - (externalPadding * 2)
+        : (linden.dimen.screenWidth / 3) * 2;
+
+    final container = Container(
+      constraints: BoxConstraints(maxWidth: maxWidth),
+      height: linden.dimen.unit8,
+      child: sized,
+    );
+    final padding = SizedBox(width: internalPadding);
+    return Row(
+      children: [padding, container, padding],
+      mainAxisAlignment: MainAxisAlignment.center,
+    );
+  }
+
+  Widget _buildItem(NavBarItem data) {
+    late final Widget child;
+    switch (data.runtimeType) {
+      case NavBarItemIconButton:
+        child = NavBarIconButton(data as NavBarItemIconButton);
+        break;
```

Comment:
💡 To make this a bit more modulare adding a builder method to the navBarItem could help a lot.

Reply:
with the builder we can not make sure about proper sizes, similar look, spaces in between, etc

Reply:
discussed. agreed to left it as it is right now